### ISSUE TITLE
images not showing in dev

### ISSUE THREAD

**Issue by ecurrano** [paleobot/pbot-dev#321]
In the OTU results page, the image display seems to be broken. These images were direct uploaded to PBot (i.e., not web links).
<img width="678" alt="Screen Shot 2024-11-14 at 2 04 53 PM" src="https://github.com/user-attachments/assets/e32037a3-9ff2-4b25-9942-9988263e70ea">

**Comment by aazaff** [comment]
Unclear if this is or is not related to the below email exchange (that was not entered as an issue, bad @doricon!). My sense is Dori's view is that image linking straight-up does not work, but I don't know if that has been practicallyconfirmed.

"... I tried to figure out how to link an image [from morphosource] but have not had success. (1) I can't figure out the appropriate link in morphosource record to use, and (2) even if I use an original media source link, the image doesn't show up in PBot.  

See this specimen: [http://128.196.12.150/query/specimen/ac3da65f-e673-4eaf-83c9-473abb62cf3f?includeImages=true&includeDescriptions=true&includeOTUs=true](https://url.usb.m.mimecastprotect.com/s/MkA1CXYBgvfX98mK1c9u6FWwGVo?domain=128.196.12.150)
Morphosource still image that I tried to use: 
[https://www.morphosource.org/concern/media/000660156?locale=en](https://url.usb.m.mimecastprotect.com/s/bysUCYVDjwfL9n5lQc3CGFxU3UQ?domain=morphosource.org)
Original file from Harvard that I linked, but the image doesn't show: [https://mczbase.mcz.harvard.edu/media/3874424](https://url.usb.m.mimecastprotect.com/s/Zq_yCZZEkxi5jkzgEhNFRFBzUfc?domain=mczbase.mcz.harvard.edu)

**Comment by NoisyFlowers** [comment]
The image in question is has a link of http://128.196.196.17/images/ac3da65f-e673-4eaf-83c9-473abb62cf3f/IMG_0043editCropPbot.jpg. This is the old IP of that machine. 

There does not appear to be any way to edit this link through our current interface. We'll have to fix that.

In the meantime, I could try a global substitution to fix these IP addresses in existing Image nodes. If only I had access to that machine.

**Comment by NoisyFlowers** [comment]
I ran the following cypher on dev:
```
match 
    (i:Image)
where
    i.link =~ '.*128.196.196.17.*'
set
    i.link = apoc.text.replace(i.link, '128.196.196.17', '128.196.12.150')
return 
    i
```

Images are behaving properly now.

However, **we should keep this issue open**. We need to provide a way for a user to edit a link that becomes unreachable.

**Comment by NoisyFlowers** [comment]
To be clear, in the specific instance described above, the address of our machine changed, causing all the image urls to be wrong. That's on us, if that ever happens on the production machine, and we shouldn't expect the users to have to fix it. 

In the instance in which an image is hosted somewhere else and the link for that changes, users are currently able to edit that Image node. However, selecting it from the dropdown is difficult because it does not display anything identifiable. This is what we need to fix.

**Comment by NoisyFlowers** [comment]
I've updated Image select dropdown to also display the image caption. This will help users find the image they want to edit even if its thumb doesn't display (e.g. for editing a broken link).

I'm leaving the rest of the functionality as-is. A user who wants to edit a link will have to delete the existing link and re-enter. It's not friendly, but it works. Anything more will take more time than we should give this right now. I've opened a new issue to track whatever else we decide to do: https://github.com/paleobot/pbot-dev/issues/324

This is in
paleobot/pbot-client@df8192e12bf07c2b7d3acc0a4845b14fed320bb6

Merged to master and deployed on dev and stage.

We can close this now.
